snap/snapcraft.yaml

```diff
@@ -0,0 +1,31 @@
+name: prometheus-juju-backup-all-exporter
+base: core22
+adopt-info: prometheus-juju-backup-all-exporter
+summary: collects backup results and exports them as metrics
+description: |
+  The prometheus-juju-backup-all-exporter is a snap for collecting backup
+  results from charm-juju-backup-all, and export those results as prometheus
+  metrics. The metrics are expected to be used with prometheus.
+
+grade: stable
+confinement: strict
+
+apps:
+  prometheus-juju-backup-all-exporter:
+    daemon: simple
+    install-mode: disable
+    restart-condition: on-abnormal
+    command: bin/prometheus-juju-backup-all-exporter
+    plugs:
+      - network
```

Comment:
Does the exporter need to initiate outgoing network connections? If not, the `network-bind` should be enough to operate as a listening network service.

Reply:
I am using [this library](https://charmhub.io/operator-libs-linux/libraries/snap) to control snap lifecycle. The library might need to make request to `snapcraft.io` to get snap information online.

Reply:
These plugs affect only the process within the snap. Processes outside, like a charm triggering snap update, are not affected. So unless something in your python code will be initiating network connections then this plug is not needed.

Reply:
Thanks, I think you are right, the exporter most likely don't need to initiate network connection to the outside. Let me test it and update the code.